e2e/trust/sign_test.go

```diff
@@ -0,0 +1,80 @@
+package trust
+
+import (
+	"github.com/gotestyourself/gotestyourself/icmd"
+	"github.com/stretchr/testify/assert"
+	"os"
+	"testing"
+)
+
+const notaryURL = "https://notary-server:4443"
+const registryPrefix = "registry:5000"
+
+const alpineImage = "registry:5000/alpine:3.6"
+const busyboxImage = "registry:5000/busybox:1.27.2"
+const localImage = "registry:5000/signlocal:v1"
+const signImage = "registry:5000/sign:v1"
+
+func TestSignWithTrustSign(t *testing.T) {
+	setupConfigFile(t)
+	defer removeConfigFile(t)
+	icmd.RunCmd(icmd.Command("docker", "pull", alpineImage)).Assert(t, icmd.Success)
+	icmd.RunCommand("docker", "tag", alpineImage, signImage).Assert(t, icmd.Success)
+	result := icmd.RunCmd(
+		icmd.Command("docker", "-D", "trust", "sign", signImage),
+		withTrustAndPassphrase("root_password", "repo_password"))
+	result.Assert(t, icmd.Success)
+}
+
+func TestSignWithTrustSignlocal(t *testing.T) {
+	setupConfigFile(t)
+	defer removeConfigFile(t)
+	icmd.RunCmd(icmd.Command("docker", "pull", alpineImage)).Assert(t, icmd.Success)
+	icmd.RunCommand("docker", "tag", alpineImage, localImage).Assert(t, icmd.Success)
+	result := icmd.RunCmd(
+		icmd.Command("docker", "-D", "trust", "sign", localImage),
+		withTrustAndPassphrase("root_password", "repo_password"))
+	result.Assert(t, icmd.Success)
+	icmd.RunCommand("docker", "tag", busyboxImage, localImage).Assert(t, icmd.Success)
+	result = icmd.RunCmd(
+		icmd.Command("docker", "trust", "sign", "--local", localImage),
+		withTrustAndPassphrase("root_password", "repo_password"))
+	result.Assert(t, icmd.Success)
+}
+
+func withTrustAndPassphrase(rootPwd, repositoryPwd string) func(cmd *icmd.Cmd) {
+	return func(cmd *icmd.Cmd) {
+		env := append(os.Environ(),
+			"DOCKER_CONTENT_TRUST_SERVER="+notaryURL,
+			"DOCKER_CONTENT_TRUST_ROOT_PASSPHRASE="+rootPwd,
+			"DOCKER_CONTENT_TRUST_REPOSITORY_PASSPHRASE="+repositoryPwd,
+		)
+		cmd.Env = append(cmd.Env, env...)
+	}
+}
+
+func setupConfigFile(t *testing.T) {
+	err := os.Mkdir("/root/.docker", os.FileMode(0700))
```

Comment:
Someone might run these tests outside of a container (even though they shouldn't) so we should use a temporary directory so that we don't clobber their config.

`fs.NewDir()` (https://godoc.org/github.com/gotestyourself/gotestyourself/fs) should let you create this easily and `fs.WithMode()` should let you set the mode.

This method can return the `fs.Dir` so you can `defer dir.Remove() to cleanup, and `dir.Path()` to get the path for the temporary directory.


The file in the directory can be created using `fs.WithFile("config.json", `....`)`

Reply:
I guess this failed on CI as well because some other test created this directory already